crates/typst/src/eval/value.rs

```diff
@@ -69,6 +71,43 @@ pub enum Value {
     Dyn(Dynamic),
 }
 
+impl Serialize for Value {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Value::None => serializer.serialize_none(),
+            Value::Auto => serializer.serialize_str("auto"),
+            Value::Bool(v) => serializer.serialize_bool(*v),
+            Value::Int(v) => serializer.serialize_i64(*v),
+            Value::Float(v) => serializer.serialize_f64(*v),
+            Value::Str(v) => serializer.serialize_str(v),
+            Value::Bytes(v) => serializer.serialize_bytes(v),
+            Value::Array(v) => {
+                let mut seq = serializer.serialize_seq(Some(v.len()))?;
+                for e in v {
+                    seq.serialize_element(e)?;
+                }
+                seq.end()
+            }
+            Value::Dict(v) => {
+                let mut map = serializer.serialize_map(Some(v.len()))?;
+                for (k, v) in v {
+                    map.serialize_entry(k.as_str(), v)?;
+                }
+                map.end()
+            }
+            Value::Content(v) => serializer.serialize_str(&v.plain_text()), // Could be rendered as list of elements??
```

Comment:
As said above, I think this should serialize somewhat like your `SelectedElem` was before and both should use a shared `Serialize` impl in `Content`.

Reply:
Yes!